### ISSUE TITLE
Track Public API Issues On Flatten

### ISSUE THREAD

**Issue by allen-adastra** [allen-adastra/xarray_jax#15]
On flattening, would be ideal to use the public API to extract data, but that doesn't seem to work in some cases. Tracking the cases and reasons here:

**Public API Works**
- `xr.Variable.dims`
- `xr.Variable.attrs`
- `xr.DataArray.variable`
- `xr.DataArray.name`
- `xr.Dataset.dims`
- `xr.Dataset.attrs`

**Public API Does Not Work**
- `xr.Variable.data` breaks tree masking (e.g. `jax.tree.map(lambda _: bool, tree)`)
- `xr.DataArray.coords` Root Cause TBD
- `xr.DataArray.indexes` Root Cause TBD
- `xr.Dataset.coords` Root Cause TBD
- `xr.Dataset.indexes` Root Cause TBD
